.github/workflows/sync-upstream-pr.yml

```diff
@@ -0,0 +1,122 @@
+name: Sync with Upstream (create)
+
+on:
+  schedule:
+    - cron: "0 0 1 * *" # Runs at 00:00 on the first day of every month
+  workflow_dispatch:
+  # TODO: Remove this trigger once the workflow is ready
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.user.login != 'gha-runner-images-updater-test[bot]'
+    steps:
+      - name: Generate GitHub App token
+        id: generate-token
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ secrets.GH_APP_ID }}
+          private-key: ${{ secrets.GH_APP_KEY }}
+
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ steps.generate-token.outputs.token }}
+          ref: main
+
+      - name: Create new branch from main
+        run: |
+          git checkout -b update-from-upstream-${{ github.run_number }}
+
+      - name: Configure Git
+        run: |
+          git config user.name 'gha-runner-images-updater-test[bot]'
+          git config user.email '172421971+gha-runner-images-updater-test[bot]@users.noreply.github.com@github.com'
+
+      #TODO: Remove this step once the workflow is ready
+      - name: Merge the feature branch for the workflow
+        run: |
+          git merge --squash origin/gc/sync-upstream
+          git commit -m "Add sync-upstream.yml workflow (#7)"
+
+      - name: Pull changes from upstream and rebase
+        id: rebase
+        run: |
+          git remote add upstream https://github.com/actions/runner-images.git
+          git fetch upstream
+          git rebase upstream/main || echo "conflicts=true" >> $GITHUB_OUTPUT
```

Comment:
We will need to discuss this a little further, I'm not sure if the effort to keeping a single merge commit will reduce conflict resolution, I'm trying to investigate.

In the end the number of conflicts will be the same, you will just be resolving them in one go instead of in multiple steps, for example, 4 commits with 5 conflicts each would result in 20 conflicts to resolve, instead of tackling it 5 by 5 you would do it all at once, I'm not sure that would make it easier and could cause some confusion when trying to compare changes.

I'm not sure it's worth it to introduce a second branch, change how PRs are merged and "nuke" or main commit history just to change how conflcits are resolved while keeping the same amount of conflicts to resolve.